### PR TITLE
[loki] Allow to define existing secret for Loki configuration

### LINKED ISSUE #158: Add option to define existing secret for Loki configuration
**Is your feature request related to a problem? Please describe.**
If the administrator handles application on cluster in GitOps way, then it is not good practise to commit unencrypted password into version control but the current version of the chart does not support to reference pre-existing secret.

Given:  Administrator has Object Storage, Amazon S3 compatible. 
Then: Administrator creates `values.yaml` (snippet).
```yaml
loki:
  config:
    storage_config:
      aws:
        s3: https://minio_admin:mypassword@minio.foo.bar./loki
```
Then: Administrator commits into version control.
Then: Minio logon is credential leaked via Helm Chart parameters.

**Describe the solution you'd like**
Given:  Administrator has Object Storage, Amazon S3 compatible. 
Then: Administrator creates secret called `my-loki-secret`.
Then: Administrator creates `values.yaml` (snippet).

```yaml
loki:
  config:
    existingSecret: my-loki-secret
```
Then: Minio logon credential is not leaked via Helm Chart parameters.

**Describe alternatives you've considered**
This proposal add alternative solution to the current one.

**Additional context**
A lots of popular chart provide option to define pre-existing secret to set it as reference for application.

xref https://github.com/grafana/loki/issues/3042

### PR DESCRIPTION
**Pull request by alex1989hu**
**What this PR does / why we need it**:
Add alternative solution to let users provide secrets which holds Loki configuration.
**Which issue(s) this PR fixes**:
Fixes #158

Signed-off-by: Alex Szakaly <alex.szakaly@gmail.com>

### CODE REVIEW

**Review by haim-ari — COMMENTED** [review]
Please approve and pull.
This will allow to use Sealed Secrets + Flux2 HelmRelease to deploy Loki.

**Comment by haimari** [comment]
@alex1989hu Hello,
Can you please merge ?

**Comment by alex1989hu** [comment]
> @alex1989hu Hello,
> Can you please merge ?

Hello, I do not have write permission, but based on recent PRs: @zanhsieh has.

**Comment by haimari** [comment]
@zanhsieh , can you please check ?

File: charts/loki/values.yaml [hunk 1]
```diff
@@ -44,6 +44,8 @@ tracing:
   jaegerAgentHost:
 
 config:
+  ## Name of the existing secret containing loki.yaml key and the configuration as value
+  # existingSecret:
```
Comment (torstenwalter) [inline charts/loki/values.yaml:48]: Let's remove the comment here. That helps IDEs to do auto completion.

**Review by torstenwalter — COMMENTED** [review]
Just a small remark PR looks good for me.

File: charts/loki/values.yaml (hunk 1 above)
Comment (alex1989hu) [reply charts/loki/values.yaml:48]: https://github.com/grafana/helm-charts/pull/542

**Comment by alex1989hu** [comment]
Reopened here https://github.com/grafana/helm-charts/pull/542

**Pull request closed without merging**